File: src/test/java/com/kobylynskyi/graphql/codegen/kotlin/compile-pass/EventPropertyChildParametrizedInput.kt
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLParametrizedInput
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer
import java.util.StringJoiner
/**
 * Parametrized input for field child in type EventProperty
 */
data class EventPropertyChildParametrizedInput(
    val first: Int?,
    val last: Int?
) : GraphQLParametrizedInput {

    override fun deepCopy(): EventPropertyChildParametrizedInput {
        return EventPropertyChildParametrizedInput(
            this.first,
            this.last
        )
    
    }

    override fun toString(): String {
        val joiner = StringJoiner(", ", "( ", " )")
        if (first != null) {
            joiner.add("first: " + GraphQLRequestSerializer.getEntry(first))
        }
        if (last != null) {
            joiner.add("last: " + GraphQLRequestSerializer.getEntry(last))
        }
        return joiner.toString()
    }
}

File: src/test/java/com/kobylynskyi/graphql/codegen/scala/compile-pass/EventPropertyChildParametrizedInput.scala
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLParametrizedInput
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer
import scala.collection.JavaConverters._

/**
 * Parametrized input for field child in type EventProperty
 */
case class EventPropertyChildParametrizedInput(
    first: scala.Option[Int],
    last: scala.Option[Int]
) extends GraphQLParametrizedInput {

    override def deepCopy(): EventPropertyChildParametrizedInput = {
        EventPropertyChildParametrizedInput(
            this.first,
            this.last
        )
    }

    override def toString(): String = {
        scala.Seq(
            if (first.isDefined) "first: " + GraphQLRequestSerializer.getEntry(first.get) else "",
            if (last.isDefined) "last: " + GraphQLRequestSerializer.getEntry(last.get) else ""
        ).filter(_ != "").mkString("(", ",", ")")
    }

}
